docs/contracts.rst

```diff
@@ -746,39 +746,49 @@ converted to ``uint8``.
-in the blockchain. These logs are associated with the address of
-the contract and will be incorporated into the blockchain
-and stay there as long as a block is accessible (forever as of
-Frontier and Homestead, but this might change with Serenity). Log and
-event data is not accessible from within contracts (not even from
-the contract that created them).
-
-SPV proofs for logs are possible, so if an external entity supplies
-a contract with such a proof, it can check that the log actually
-exists inside the blockchain.  But be aware that block headers have to be supplied because
-the contract can only see the last 256 block hashes.
-
-Up to three parameters can
-receive the attribute ``indexed`` which will cause the respective arguments
-to be stored in a special data structure as so-called "topics", which allows them to be searched for,
-for example when filtering a sequence of blocks for certain events. Events can always
-be filtered by the address of the contract that emitted the event. Also,
-the hash of the signature of the event is one of the topics except if you
-declared the event with ``anonymous`` specifier. This means that it is
-not possible to filter for specific anonymous events by name.
-
-If arrays (including ``string`` and ``bytes``) are used as indexed arguments, the
-Keccak-256 hash of it is stored as topic instead. This is because a topic
-can only hold a single word (32 bytes).
+in the blockchain. These logs are associated with the address of the contract,
+are incorporated into the blockchain, and stay there as long as a block is
+accessible (forever as of the Frontier and Homestead releases, but this might
+change with Serenity). The Log and its event data is not accessible from within
+contracts (not even from the contract that created them).
+
+It is possible to request a simple payment verification (SPV) for logs, so if
+an external entity supplies a contract with such a verification, it can check
+that the log actually exists inside the blockchain. You have to supply block headers
+because the contract can only see the last 256 block hashes.
+
+You can add the attribute ``indexed`` to up to three parameters which adds them
+to a special data structure known as :ref:`"topics" <events_topics>`. If you use
+arrays (including ``string`` and ``bytes``) as indexed arguments, its Keccak-256
+hash is stored as a topic instead, this is because a topic can only hold a single word (32 bytes).
+
+All parameters without the ``indexed`` attribute are :ref:`ABI-encoded <ABI>`
+into the data part of the log.
+
+Topics allow you to search for events, for example when filtering a sequence of
+blocks for certain events. You can also filter events by the address of the
+contract that emitted the event.
+
+For example, the code below uses the JSON RPC ``eth_getFilterChanges``
+`method <https://github.com/ethereum/wiki/wiki/JSON-RPC#returns-42>`_ to filter
+logs that match a topic with a certain address value:
```

Comment:
I think this is still too cryptic. Also, it should not use `...` unless it is clear that those are not part of the actual string.

Reply:
OK…

1. I see `…` used in a lot of place in the solidity docs, are you saying that all those instances are wrong, or is there a subtle difference of when to use it? It's fairly standard syntax to use, but I admit, I have never used it in the middle of a string before. But, other examples in the docs do. A placeholder might be a more readable solution.
2. I need a little help here, what do you mean by 'cryptic'? Maybe to add an example of logs the filter returns?

Reply:
the example does not use json-rpc. It is just a snippet of the payload of a json-rpc request. It misses the function name and other things. If we do this, we should explain json-rpc a bit more, but I think it would be better to use web3.js instead.

Reply:
Why not provide the full string? Also, I don't think we have used the single-character unicode `...` before in these docs.

Reply:
Here's the relevant part of web3.js docs I believe https://web3js.readthedocs.io/en/1.0/web3-eth-subscribe.html#subscribe-logs

Reply:
OK all, try this change. I'm still learning a lot here, so I hope it's at least partially right.

@chriseth regarding '…' vs '...' (😆), I used to work in print, so have aways used the 'correct one out of habit, but I'm really not bothered about using it. For unicode reasons I will stick to the 3 dots from now on.